src/smoke-tests/smoke-tests.js

```diff
@@ -0,0 +1,121 @@
+const { spawnSync } = require('child_process');
+const crypto = require('crypto');
+const fs = require('fs-extra');
+const os = require('os');
+const path = require('path');
+
+require('should');
+const fetch = require('node-fetch');
+
+describe('smoke tests - going to take some time', () => {
+  const context = {
+    package: {
+      filename: null,
+      version: null,
+      path: null
+    },
+    workdir: null,
+    cliBin: null
+  };
+
+  before(() => {
+    if (process.env.DEPLOY_KEY) {
+      const rcPath = path.join(os.homedir(), '.zapierrc');
+      if (!fs.existsSync(rcPath)) {
+        fs.writeFileSync(
+          rcPath,
+          JSON.stringify({ deployKey: process.env.DEPLOY_KEY })
+        );
+      }
+    }
+
+    const npmPack = spawnSync('npm', ['pack'], { encoding: 'utf8' });
+    const lines = npmPack.stdout.split('\n');
+    for (let i = lines.length - 1; i >= 0; i--) {
+      const line = lines[i].trim();
+      if (line) {
+        context.package.filename = line;
+        break;
+      }
+    }
+    context.package.version = context.package.filename.match(
+      /\d+\.\d+\.\d+/
```

Comment:
might be worth moving this regex into a constant so it can be enforced the same way across the package (since I bet we use the same pattern elsewhere)

Reply:
Made it a constant in 00f2354. I couldn't find anywhere else in the cli repo that uses this regex. So I think we can keep it in this file (`smoke-tests.js`) for now.